src/App.tsx

```diff
@@ -1,26 +1,84 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import { FC, useEffect, useMemo, useState } from 'react';
 
-const USER_ID = 0;
+import { Todo } from './types/Todo';
+import { Errors } from './types/Errors';
+import { FilterBy } from './types/FilterBy';
 
-export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+import { getTodos } from './api/todos';
+import { getFilteredTodos } from './utils/getFilteredTodos';
+import { handleError } from './utils/handleError';
+import { handleDeleteTodos } from './utils/handleDeleteTodos';
+
+import { Header, TodoList, Footer, ErrorMessage, TodoItem } from './components';
+
+export const App: FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [error, setError] = useState<Errors>(Errors.DEFAULT);
+  const [idsForDelete, setIdsForDelete] = useState<number[]>([]);
+  const [selectedFilter, setSelectedFilter] = useState<FilterBy>(FilterBy.ALL);
+
+  const completedTodosId = useMemo(() => {
+    return getFilteredTodos(todos, FilterBy.COMPLETED).map(todo => todo.id);
+  }, [todos]);
+
+  const numberOfActiveTodos = useMemo(() => {
+    return getFilteredTodos(todos, FilterBy.ACTIVE).length;
+  }, [todos]);
+
+  const filteredTodos = useMemo(() => {
+    return getFilteredTodos(todos, selectedFilter);
+  }, [todos, selectedFilter]);
+
+  useEffect(() => {
+    if (idsForDelete.length) {
+      handleDeleteTodos(idsForDelete, setTodos, setIdsForDelete, setError);
+    }
+  }, [idsForDelete]);
+
+  useEffect(() => {
+    getTodos()
+      .then(setTodos)
+      .catch(() => handleError(Errors.LOAD_ERROR, setError));
+  }, []);
```

Comment:
It's better to move the logic inside the useEffect hook to a separate function, as it improves code readability, reusability, and makes testing easier. It also helps keep useEffect focused on side effects.